dev/run-tests

```diff
@@ -17,239 +17,394 @@
+    """Takes a string and an abritrary number of prefixes then checks the
+    original string for any of the possible prefixes passed in"""
+
+    for s in prefixes:
+        if orig_str.startswith(s):
+            return True
+    return False
+
+# This function current acts to determine if SQL tests need to be run in
+# addition to the core test suite *or* if _only_ SQL tests need to be run
+# as the git logs show that to be the only thing touched. In the future
+# this function will act more generically to help further segregate the
+# test suite runner (hence the function name).
+# @return a set of unique test names
+def determine_test_suite():
+    test_suite = list()
+
+    if amplab_jenkins:
+        sp.Popen(['git', 'fetch', 'origin', 'master:master']).wait()
+
+        raw_output = sp.check_output(['git', 'diff', '--name-only', 'master'])
+        # remove any empty strings
+        changed_files = [f for f in raw_output.split('\n') if f]
+
+        # find any sql files
+        sql_files = [f for f in changed_files
+                     if multi_starts_with(f, 
+                                          "sql/", 
+                                          "bin/spark-sql", 
+                                          "sbin/start-thriftserver.sh")]
+
+        non_sql_files = set(changed_files).difference(set(sql_files))
+
+        if non_sql_files:
+            test_suite.append("CORE")
+        if sql_files:
+            print "[info] Detected changes in SQL. Will run Hive test suite."
+            test_suite.append("SQL")
+            if not non_sql_files:
+                print "[info] Detected no changes except in SQL. Will only run SQL tests."
+        return set(test_suite)
+    else:
+        # we aren't in the Amplab environment so merely run all tests
+        test_suite.append("CORE")
+        test_suite.append("SQL")
+        return set(test_suite)
+
+def set_title_and_block(title, err_block):
+    os.environ["CURRENT_BLOCK"] = error_codes[err_block]
+    line_str = "".join(['='] * 72)
+
+    print
+    print line_str
+    print title
+    print line_str
+
+def run_cmd(cmd):
+    """Given a command as a list of arguments will attempt to execute the
+    command and, on failure, print an error message"""
+
+    if not isinstance(cmd, list):
+        cmd = cmd.split()
+    try:
+        sp.check_output(cmd)
+    except sp.CalledProcessError as e:
+        print "[error] running", e.cmd, "; received return code", e.returncode
+        exit(e.returncode)
+
+def run_apache_rat_checks():
+    set_title_and_block("Running Apache RAT checks", "BLOCK_RAT")
+    run_cmd(["./dev/check-license"])
+
+def run_scala_style_checks():
+    set_title_and_block("Running Scala style checks", "BLOCK_SCALA_STYLE")
+    run_cmd(["./dev/lint-scala"])
+
+def run_python_style_checks():
+    set_title_and_block("Running Python style checks", "BLOCK_PYTHON_STYLE")
+    run_cmd(["./dev/lint-python"])
+
+def exec_maven(mvn_args = []):
```

Comment:
Another PEP 8 nit: No spaces around the `=` when defining default or parameter values in a function call.